src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

```diff
@@ -278,6 +278,8 @@ where applicable:
 Some parametrically linked anatomical images involve both magnitude and phase  
 reconstructed images in the calculation of a parameter map. In that case, the 
 filename MUST make use of this key/value pair to distinguish between them. 
+Phase images SHOULD be in radians and have a range of (0, 2 pi]
```

Comment:
You just want to go with SHOULD here? I'd be fine with enforcing MUST if you're up for it @Gilles86....

If we go with SHOULD then we can't validate it....I think. @chrisgorgo - does the validator tell you anything about "SHOULD" requirements? Do they come up as warnings for example?

Reply:
Thinking about this: in fMRI world, people use phase images or difference-of-phase images that are definetely not always adhering to radians. Maybe keep it SHOULD? Let's discuss it later today.

Reply:
Bumping following creation of #63.

If, in a BIDS App, I wish to read magnitude & phase data, and convert such to complex data (ie. real & imaginary components), I need to know the numerical range of the phase image. This is trivial if the units of such are enforced, but ill-posed otherwise. The actual position of the interval range (or indeed enforcement of such) doesn't matter that much, since it's sine / cosine underlying that transformation; what matters is the scaling.

Forcing adherence to radians in the construction of a BIDS dataset would I think make more sense than demanding that any application reading such data in an automated fashion be capable of inferring the units / range of the input data.

Reply:
Hey Lestropie,

Thanks for your feedback! Are you saying you'd rather only enforce that the phase images are restricted to _any_ range within 0 and 2 pi, rather than specifically (0, 2pi]? So (-pi, pi]. should also be allowed?

Reply:
If I had it entirely my way, I'd say that phase images:

- MUST be stored in radians;

- SHOULD be in the range [0, 2pi).

The former is a highly function-driven restriction; whereas no robust piece of software should fall over if provided with phase data that are defined in the range [-pi, pi) rather than [0, 2pi), so it doesn't need to be regulated so strictly (unless there's a scenario I'm overlooking).

Reply:
OK. I agree with that. I think current BIDS apps like Fmriprep just assume that 0 and 2 pi are exactly in the range of raw values in an image and then do something like `(data - data.min()) / (data.max() - data.min()) * 2 * pi)`, which is a real pity.